x/twap/logic.go

```diff
@@ -103,6 +103,20 @@ func (k Keeper) updateRecords(ctx sdk.Context, poolId uint64) error {
 		return err
 	}
 
+	denoms, err := k.ammkeeper.GetPoolDenoms(ctx, poolId)
+	if err != nil {
+		return err
+	}
+
+	// given # of denoms in the pool namely, that for `k` denoms in pool,
+	// there should be k * (k - 1) / 2 records
+	denomNum := len(denoms)
+	expectedRecordsLength := denomNum * (denomNum - 1) / 2
```

Comment:
Lets add a comment for what we're doing here, or move `expectedRecordsLength := getExpectedNumRecords(ctx, pool Id)` as a separate function?